front/app/components/UI/RHFInputMultilocWithLocaleSwitcher/index.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { isNilOrError } from 'utils/helperUtils';
+import useAppConfigurationLocales from 'hooks/useAppConfigurationLocales';
+import {
+  InputMultilocWithLocaleSwitcher,
+  InputMultilocWithLocaleSwitcherProps,
+} from '@citizenlab/cl2-component-library';
+import Error from 'components/UI/Error';
+import { Controller, useFormContext, FieldError } from 'react-hook-form';
+import { Locale } from 'typings';
+
+export interface Props
+  extends Omit<
+    InputMultilocWithLocaleSwitcherProps,
+    'locales' | 'valueMultiloc'
+  > {
+  name: string;
+}
+
+const RHFInputMultilocWithLocaleSwitcher = ({ name, ...rest }: Props) => {
+  const {
+    formState: { errors },
+    control,
+  } = useFormContext();
+  const locales = useAppConfigurationLocales();
+
+  if (isNilOrError(locales)) {
+    return null;
+  }
+
+  const defaultValue: Partial<Record<Locale, any>> = locales.reduce(
+    (acc, curr) => ((acc[curr] = ''), acc),
+    {}
+  );
+
+  const errorMessage = Object.values(
```

Comment:
Same here, personal preference but maybe this could be a bit more verbose to make it more clear what's happening at first glance

Reply:
Added a comment to clarify it 👍